### PR TITLE
Add Spyglass backend

### PR DESCRIPTION
**Pull request by imphil**
This PR adds a Synopsys (formerly Atrenta) Spyglass backend to fusesoc. The general approach has been discussed in #158. As discussed there the backend is CAPI2-only.

Example use
~~~
git clone git@github.com:optimsoc/optimsoc.git
cd optimsoc
git checkout spyglass
export OPTIMSOC_SRC=$PWD

fusesoc --verbose --cores-root $OPTIMSOC_SRC run --target=synth --tool=spyglass optimsoc:examples:compute_tile_nexys4ddr_capi2
~~~

While working on this I've noticed a couple things I don't yet understand of CAPI2 or might also be limitations. I've made note of these things in https://github.com/optimsoc/optimsoc/blob/spyglass/examples/fpga/nexys4ddr/compute_tile/compute_tile_nexys4ddr_capi2.core. In this case, all dependent cores are described in CAPI1 files, only the top-level core file is written in CAPI2. 

1. I cannot use a new (and self-defined) "lint" target to run spyglass, but I must use the synth target, otherwise no RTL files of our dependencies (as defined in CAPI1 files) are found.
2. I had to "hack" into the CAPI1 parser to actually add the files to the synth backend if Spyglass is used. 

I'm not sure how much of this logic is intended, what is a bug and what is just a limitation. Maybe @olofk you can give some insight here?

### CODE REVIEW

File: edalize/spyglass.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            'has_systemVerilog' : has_systemVerilog,
+            'sanitized_goals'   : [],
+        }
+
+        self.render_template('spyglass-project.prj.j2',
+                             self.name + '.prj',
+                             template_vars)
+
+        # Create a single TCL file for each goal
+        goals = ['Design_Read'] + self.tool_options['goals']
```
Comment (olofk) [inline edalize/spyglass.py]: FuseSoC crashes if goals is left out. Add a check and print raise a RuntimeError or set a sensible default if it's required

**Comment by olofk** [comment]
> I cannot use a new (and self-defined) "lint" target to run spyglass, but I must use the synth target, otherwise no RTL files of our dependencies (as defined in CAPI1 files) are found.

Are you sure about this? I added toplevel, goals and spyglass_options from the synth target to the lint target and it looks to me like the synth files from CAPI1 cores are picked up

**Comment by imphil** [comment]
Thanks @olofk for having a look. I've uploaded v2 of the PR.
- The synth vs. lint target issue is indeed non-existent, I don't know what I did wrong when I tested previously.
- I've added default values.

The example CAPI2 file at https://github.com/optimsoc/optimsoc/blob/spyglass/examples/fpga/nexys4ddr/compute_tile/compute_tile_nexys4ddr_capi2.core has been updated to showcase these changes.

**Comment by olofk** [comment]
Looks great. I realize that I should have asked for a test case as well, but we can take that as a separate PR. Picked and pushed

**Pull request closed without merging**